File: src/main/java/committee/nova/vlzoomer/client/event/ClientHandler.java
package committee.nova.vlzoomer.client.event;

import committee.nova.vlzoomer.client.config.ClientConfig;
import committee.nova.vlzoomer.client.key.ModeKeyHandler;
import committee.nova.vlzoomer.client.key.ZoomKeyHandler;
import cpw.mods.fml.client.registry.KeyBindingRegistry;
import net.minecraft.client.settings.KeyBinding;
import net.minecraftforge.client.event.FOVUpdateEvent;
import net.minecraftforge.common.MinecraftForge;
import net.minecraftforge.event.ForgeSubscribe;
import org.lwjgl.input.Keyboard;

public class ClientHandler {
    public static void init() {
        KeyBindingRegistry.registerKeyBinding(new ZoomKeyHandler(new KeyBinding[]{new KeyBinding("key.vlzoomer.zoom", Keyboard.KEY_Z)}));
        KeyBindingRegistry.registerKeyBinding(new ModeKeyHandler(new KeyBinding[]{new KeyBinding("key.vlzoomer.switchMode", Keyboard.KEY_X)}));
        MinecraftForge.EVENT_BUS.register(new ClientHandler());
    }

    @ForgeSubscribe
    public void onFov(FOVUpdateEvent e) {
        if (!ClientConfig.isToggleMode()) {
            if (ZoomKeyHandler.hasBeenPressed()) e.newfov = e.fov * ClientConfig.getFovModifier();
            return;
        }
        e.newfov = e.fov * ClientConfig.getToggled();

    }
}

File: src/main/scala/committee/nova/vlzoomer/client/handler/ClientHandler.scala
package committee.nova.vlzoomer.client.handler

import committee.nova.vlzoomer.client.config.ClientConfig
import committee.nova.vlzoomer.client.handler.ClientHandler.ZOOM
import cpw.mods.fml.client.registry.ClientRegistry
import cpw.mods.fml.common.FMLCommonHandler
import cpw.mods.fml.common.eventhandler.SubscribeEvent
import net.minecraft.client.settings.KeyBinding
import net.minecraftforge.client.event.FOVUpdateEvent
import net.minecraftforge.common.MinecraftForge
import org.lwjgl.input.Keyboard

object ClientHandler {
  final val ZOOM: KeyBinding = new KeyBinding("key.vlzoomer.zoom", Keyboard.KEY_Z, "category.vlzoomer")
  final val SWITCH: KeyBinding = new KeyBinding("key.vlzoomer.switchMode", Keyboard.KEY_X, "category.vlzoomer")

  def init(): Unit = {
    ClientRegistry.registerKeyBinding(ZOOM)
    ClientRegistry.registerKeyBinding(SWITCH)
    MinecraftForge.EVENT_BUS.register(new ClientHandler)
    FMLCommonHandler.instance().bus().register(new FMLEventHandler)
  }
}

class ClientHandler {
  @SubscribeEvent
  def onFov(e: FOVUpdateEvent): Unit = {
    if (!ClientConfig.isToggleMode) {
      // Weird, but have to do so to avoid glitches
      if (ZOOM.getIsKeyPressed | ZOOM.isPressed) e.newfov = e.fov * ClientConfig.getHoldFovModifier
      return
    }
    e.newfov = e.fov * ClientConfig.getToggled
  }
}
